crates/net/discv5/src/lib.rs

```diff
@@ -299,19 +312,20 @@ impl Discv5 {
     }
 
     /// Tries to convert an [`Enr`](discv5::Enr) into the backwards compatible type [`NodeRecord`],
-    /// w.r.t. local [`IpMode`]. Tries the socket from which the ENR was sent, if socket is missing
-    /// from ENR.
+    /// w.r.t. local [`IpMode`]. Uses source socket as udp socket.
     ///
-    ///  Note: [`discv5::Discv5`] won't initiate a session with any peer with a malformed node
-    /// record, that advertises a reserved IP address on a WAN network.
+    ///  Note: [`discv5::Discv5`] won't initiate a session with any peer with an unverifiable node
+    /// record, for example one that advertises a reserved LAN IP address on a WAN network. This is
+    /// in order to prevent DoS attacks, where some malicious peers may advertise a victim's socket.
     pub fn try_into_reachable(
         &self,
         enr: &discv5::Enr,
         socket: SocketAddr,
     ) -> Result<NodeRecord, Error> {
         let id = enr_to_discv4_id(enr).ok_or(Error::IncompatibleKeyType)?;
 
-        let udp_socket = self.ip_mode().get_contactable_addr(enr).unwrap_or(socket);
+        // todo: when impl ipv6 support in rlpx, favour advertised socket if ipv6 wrt to local ip
+        // mode
```

Comment:
is there anything we should be tracking for this?

unclear what favoring means here in this context, would lean towards removing this.

Reply:
was thinking about dual stack here, think eventually, rlpx will have to support dual stack too: ipv6 and ipv4 sockets in parallel. consider it removed.